crates/relayer/src/foreign_client.rs

```diff
@@ -1786,13 +1789,24 @@ impl<DstChain: ChainHandle, SrcChain: ChainHandle> ForeignClient<DstChain, SrcCh
         // If the misbehaving chain is a CCV consumer chain, we need to add
         // the corresponding CCV message for the provider.
         if is_ccv_consumer_chain {
```

Comment:
Do we stll need to check this here?

Reply:
We could remove it and use the `match fetch_ccv_consumer_id()` to determine if it is a consumer chain, but that would remove logging errors when `fetch_ccv_consumer_id` fails for another reason.
So I would be of the opinion to keep this check

Reply:
That's a good point. Vaguely remember that we had a way to tell if the chain doesn't support the grpc vs other failure. But it is not a good solution either so let's keep the config flag for now.